docs-requirements.txt

```diff
@@ -6,7 +6,7 @@ sphinx-autodoc-typehints~=1.10.2
 # doesn't work for pkg_resources.
 ./opentelemetry-api
 ./opentelemetry-sdk
-./opentelemetry-python-contrib/exporter/opentelemetry-exporter-datadog
+-e "git+https://github.com/open-telemetry/opentelemetry-python-contrib.git#egg=opentelemetry-python-contrib&subdirectory=exporter/opentelemetry-exporter-datadog" --src .
```

Comment:
`-e` says install this in `editable` mode, this is required to use the `--src` flag

`--src` says install this repo from VCS at this path. In this case it will clone the contrib repo at path `.` and call it `opentelemetry-python-contrib` because of the `#egg-opentelemetry-python-contrib` parameter

`subdirectory=exporter/opentelemetry-exporter-datadog` is required because pip install will immediately try to install a package after cloning so it must be told where it can find a `setup.py` otherwise it will fail.

Subsequent installs can take advantage of the first install which already cloned the repo so we don't need to clone multiple times :smile: